### PR TITLE
build footers from json data

### PR DESCRIPTION
**Pull request by SimonLab**
see #209

### CODE REVIEW

**Comment by alanshaw** [comment]
👓

**Comment by alanshaw** [comment]
👓

File: client/routes/home.js [hunk 1]
```diff
@@ -7,11 +7,17 @@ module.exports = function (page) {
...
+    if (!ctx.isInitialRender) {
+      var pageEl = document.getElementsByTagName('main')[0];
+      data.footer = require('../../fixtures/footer');
+      data.footerBanner = require('../../fixtures/footer-banner');
+      pageEl.innerHTML = Templates['home'](data);
+      // refresh the title of the page
+      document.getElementsByTagName('title')[0].textContent = data.titlePage;
+      next();
+    } else {
+      listeners();
```
Comment (alanshaw) [inline client/routes/home.js]: This should be a call to next also, so we don't introduce a bug in the future when the `listeners` function also needs to use `ctx`.

**Comment by alanshaw** [comment]
@SimonLab I'm not sure if the requirements have changed but as far as I can see this PR only allows the footer links to be easily configured to point at external URLs - in @jamieu's [last comment](https://github.com/TheScienceMuseum/collectionsonline/issues/209#issuecomment-236546216) he was looking for a combination of the two - a on-site page (and markdown based page text) for each footer item that can be overridden by an external URL if set.

Could you please let me know if this PR is meant to resolve the issue (because requirements have changed) or if it is simply a step towards completing it?

**Comment by SimonLab** [comment]
At the moment we don't have any content or template for the pages linked on the footer. At the moment each link on the footer are described on the footer.js file and each link is defined with the url property. This url can target the future page on the website or an external page.
So I think I just need to have the list of external links otherwise a template to create the content of the pages, is this right or I missed some details of the feature?

**Comment by alanshaw** [comment]
@SimonLab no thats all good - we're waiting on content...

**Pull request merged**